hack/prepare-release.sh

```diff
@@ -0,0 +1,161 @@
+            ;;
+        h)  usage
+            exit 0
+            ;;
+        *)  usage
+            exit 1
+            ;;
+    esac
+done
+shift "$((OPTIND - 1))"
+
+# Check that no extra args were provided
+if [ $# -ne 1 ]; then
+    if [ $# -lt 1 ]; then
+        echo -e "ERROR: too few arguments\n"
+    else
+        echo -e "ERROR: unknown arguments: ${@:3}\n"
+    fi
+    usage
+    exit 1
+fi
+
+if [ -z "$no_patching" -a -z "$golang_version" ]; then
+    echo -e "ERROR: '-g GOLANG_VERSION' must be specified when modifying repo (i.e. when '-a' is not used)\n"
+    usage
+    exit 1
+fi
+
+release=$1
+shift 1
+
+container_image=nvcr.io/nvidia/k8s-device-plugin:$release
+
+#
+# Check/parse release number
+#
+if [ -z "$release" ]; then
+    echo -e "ERROR: missing RELEASE_VERSION\n"
+    usage
+    exit 1
+fi
+
+if [[ $release =~ ^(v[0-9]+\.[0-9]+)(\..+)?$ ]]; then
+    docs_version=${BASH_REMATCH[1]}
+    semver=${release:1}
+else
+    echo -e "ERROR: invalid RELEASE_VERSION '$release'"
+    exit 1
+fi
+
+#
+# Modify files in the repo to point to new release
+#
+if [ -z "$no_patching" ]; then
+    # Patch versions.mk
+    echo Patching versions.mk to refer to $release
+    sed -i.bak "s/^VERSION.*$/VERSION ?= $release/" versions.mk
+
+    # Patch deployments/static
+    echo Patching deployments to refer to $semver
+    find deployments/static -name "*.yaml" -exec sed -E -i '' \
+        -e s",^([[:space:]]+)app.kubernetes.io\/version:.+$,\1app.kubernetes.io\/version: $semver," \
+        -e s",^([[:space:]]+)- image:.+$,\1- image: $container_image," \
+        {} +
+    
+    # Patch deployments/helm/Chart.yaml
+    echo Patching deployments/helm/Chart.yaml to refer to $semver
+    sed -i.bak "s/^version: .*/version: \"$semver\"/" deployments/helm/nvidia-device-plugin/Chart.yaml
+    sed -i.bak "s/^appVersion: .*/appVersion: \"$semver\"/" deployments/helm/nvidia-device-plugin/Chart.yaml
+
+    # Patch README.md
+    # NOTE: this needs to be updated once we start releasing x.2.x versions
+    echo Patching README.md to refer to $release
+    sed -E -i.bak -e "s/v0\.[15-9]+\.[0-9]+/$release/g" README.md
+    sed -E -i.bak -e "s/0\.[15-9]+\.[0-9]+/$semver/g" README.md
+
+    # rm backup files
+    find . -name "*.bak" -exec rm {} \;
+fi
+
+#
+# Create release assets to be uploaded
+#
+if [ -z "$no_assets" ]; then
+    devicePlugin_chart_name="nvidia-device-plugin-chart-$semver.tgz"
+    gfd_chart_name="gpu-feature-discovery-chart-$semver.tgz"
+
+    helm package deployments/helm/nvidia-device-plugin/ --version $semver
+    mv nvidia-device-plugin-$semver.tgz $devicePlugin_chart_name
+
+    create_gfd_helm_chart
+    helm package deployments/helm/gpu-feature-discovery/ --version $semver
+    mv gpu-feature-discovery-$semver.tgz $gfd_chart_name
+    
+    cat << EOF
+
+*******************************************************************************
+*** Please manually upload the following generated files to the Github release
```

Comment:
We don't currently upload the helm packages to the release page. Is this something that we want to start doing?

Reply:
is a suggestion, some upstream projects do it. if you don't agree I can remove

Reply:
I'm happy to do it, but I don't want to do it manually.

Reply:
I have an idea, will create a follow up PR